.github/workflows/validate.yaml

```diff
@@ -0,0 +1,112 @@
+name: VALIDATE
+
+on:
+  push:
+  pull_request:
+  schedule:
+    - cron:  '0 0 * * SUN'
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        BUILD_TYPE: [Release, Debug]
+
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Dependencies
+        run: sudo apt-get update && sudo apt-get install -yq openmpi-bin libfftw3-dev python3-dev libboost-all-dev python3-h5py
+
+      - name: Configure
+        run: cmake -S ${{github.workspace}} -B ${{github.workspace}}/build -DCMAKE_BUILD_TYPE=${{ matrix.BUILD_TYPE }} -DEXTERNAL_MPI4PY=OFF
+
+      - name: Build
+        run: cmake --build ${{github.workspace}}/build --target all -j 2
+
+      #    - name: Install
+      #      run: cmake --install ${{github.workspace}}/build
+
+      - name: Test
+        working-directory: ${{github.workspace}}/build
+        run: ctest --rerun-failed --output-on-failure -j 2
+
+  documentation:
+    runs-on: ubuntu-latest
+    environment: CI
+    steps:
+      - name: install dependencies
+        # very ugly but did not get line breaks working...
+        run: sudo apt-get update && sudo apt-get install -yq openmpi-bin libfftw3-dev python3-dev libboost-all-dev python3-h5py doxygen sphinx-common graphviz latexmk texlive-latex-base texlive-latex-extra texlive-latex-recommended ghostscript
+
+      - uses: actions/checkout@v2
+        with:
+          path: 'espressopp'
+
+      - name: build documentation
+        run: |
+          cmake -S espressopp -B espressopp-build
+          cmake --build espressopp-build --target doxygen sphinx sphinx-pdf -j 2
+
+      - uses: actions/checkout@v2
+        with:
+          repository: 'espressopp/espressopp.github.io.git'
+          ref: 'master'
+          path: 'html'
+
+      - name: update docu repo
+        working-directory: ${{github.workspace}}/html
+        run: |
+          rm -rf *
+          mv ../espressopp-build/doc/sphinx/html/* .
+          mv ../espressopp-build/doc/sphinx/pdf/ESPResSo++.pdf .
+
+      - name: commit
+        working-directory: ${{github.workspace}}/html
+        run: |
+          git config --global user.name "Automatic Deployment (Travis CI)";
+          git config --global user.email "espp-devel@listserv.mpip-mainz.mpg.de";
+          git add --all && git commit -m "Documentation Update";
+
+      - name: register deploy key
+        env:
+          SSH_AUTH_SOCK: /tmp/ssh_agent.sock
+        run: |
+          ssh-agent -a $SSH_AUTH_SOCK > /dev/null
+          ssh-add - <<< "${{ secrets.PAGES_KEY }}"
```

Comment:
Is there any advantage of `ssh-agent` over just put it in a file?

Reply:
It got it working faster than the other way ;) I do not think there is much difference.